src/models/user.ts

```diff
@@ -32,23 +32,39 @@ userSchema.methods.checkPass = async function (password:string) {
 const Profile = mongoose.model(
     'Profile',
     new Schema({
-        firstName: {
+        email: {
             type: String,
             required: true,
         },
+        firstName: {
+            type: String,
+        },
         lastName: {
             type: String,
-            required: true,
         },
         address: {
             type: String,
         },
+        city: {
+            type: String,
+        },
+        country: {
+            type: String,
+        },
+        phoneNumber: {
+            type: String,
+        },
+        biography: {
+            type: String,
+        },
+        avatar: {
+            type: String
+        },
         user: {
             type: mongoose.Types.ObjectId,
             ref: 'User',
-            required: true,
             unique: true,
```

Comment:
Make this field `required` to avoid creating `profiles` that are not linked to any existing user. And we can also take advantage of the `mongoose` events triggers to do database cleanups. This will requires a change in models definition where we separate the models from schemas (At least to my knowledge)

Example

```
// Delete related profile when a user is deleted to keep a clean relationship
const userSchema = new Schema({

// add all existing fields definitions
})
userSchema.pre("remove", async function (){
await Profile.findAndRemove({where: {_id: this.profile})
})
// this solution can be customized, even be used to automatically create a profile on user creation
// you can use userSchem.post("save", callback)

```